Message privacy (#3015)
Nexmo/nexmo-developer: _documentation/en/messaging/sms/message-privacy.md

---
title: Message Privacy
---

# Message Privacy

Vonage's Auto-redact feature enables customers to protect their SMS privacy by replacing all Personally Identifiable Information (PII) in Vonage logs with the string "REDACTED". Auto-redact is automatic and is suitable for cases when you don't want sensitive customer information, like SMS message content, to end up in Vonage logs.

In general, businesses dealing with industries and sectors that are highly regulated, such as financial services, public authorities, educational institutions, hospitals, and the police need to safeguard sensitive data to comply with data protection laws and regulations.

## How does Auto-redact work?

Redaction is the editing of a document or a record to delete or mask Personally Identifiable Information (PII) or confidential information. For example, your telephone number, bank details, or your address.

When you send SMS messages using the Vonage SMS API, two pieces of PII get recorded: the SMS message body content and the receiver's phone number. When you use Vonage SMS API to receive SMS, we record the SMS message body content and the sender's phone number.

> This sensitive PII ends up in two places: server logs and transactional records of the activity. Server logs are retained for around 15 days, no more than one month. Transactional records of the activity, however, are stored for 13 months. Transactional records of the activity are known as Call Detail Records (CDRs). Both server logs and CDRs can be viewed by Vonage support staff, for testing, debugging, diagnosing user issues and reconciling CDRs against customer's transaction records.

Reviewer: ```suggestion
This sensitive PII ends up in two places: server logs and transactional records of the activity. Server logs are retained for around 15 days, no more than one month. Transactional records of the activity, however, are stored for 13 months. Transactional records of the activity are known as Call Detail Records (CDRs). Both server logs and CDRs can be viewed by Vonage support staff, for testing, debugging, diagnosing user issues and reconciling CDRs against customers' transaction records.
```

The image below shows an example of the flow of Advanced SMS Auto-redact with Vonage.

![Advanced SMS Redaction](/assets/images/messaging/sms/advanced_sms_redaction.png)

> **NB** Advanced Auto-redact works only for messages sent out using the Vonage SMS API. For messages received using the the Vonage SMS API, we can offer the Standard Auto-redact feature. For more information on the Standard Auto-redact please refer to the [Redact API documentation](https://developer.nexmo.com/redact/overview).

Reviewer: ```suggestion
**Note**: Advanced Auto-redact works only for messages sent using the Vonage SMS API. For messages received using the Vonage SMS API, we can offer the Standard Auto-redact feature. For more information on the Standard Auto-redact please refer to the [Redact API documentation](/redact/overview).
```

> Advanced Auto-redact feature ensures that the message content (SMS text) gets redacted before it gets written to the server logs and CDRs. The phone number cannot be immediately redacted in the server logs due to data retention regulations, thus we provide an option to encrypt it.

Reviewer: ```suggestion
Advanced Auto-redact feature ensures that the message content (SMS text) is redacted before it is written to the server logs and CDRs. The phone number cannot be immediately redacted in the server logs due to data retention regulations, thus we provide an option to encrypt it.
```

The redaction options available are:

1. Message content redaction only.
2. Phone number redaction only (implies encryption of phone numbers in the server logs and redaction in the long-term storage).
3. Phone number encryption only.
4. Message content redaction combined with either redaction or encryption of the phone number.

> A key point of Vonage's immediate Advanced Auto-redact is that the message body is not stored in any of the logs or the CDRs. Phone numbers are also immediately encrypted, with an encryption key that Vonage staff do not have access to. Those encrypted numbers get then redacted in the long-term storage, and the server logs expire on their own. Please refer to the (Redaction page)[https://developer.nexmo.com/redact/overview] for more information.

Reviewer: Do we mean that the encrypted numbers are redacted in the CDRs (which are stored for 13 months), but not in the server logs (which are stored from 15 days to one month). If so, can we say that? I find the references to 'long-term storage' and 'expires on their own' a bit confusing :/

Reviewer 2: Yes, with the immediate number redaction, encrypted numbers are redacted in the CDRs but not in the server logs.

Author: @obodriagovatvonagedotcom  is that fine to list out on this page?

Author: @marklewin , please let me know what you think of the revised.

---
title: Message Privacy
---

# Message Privacy

> Vonage's Auto-redact feature enables customers to protect their SMS privacy. Auto-redact automatically replaces all Personal Identifiable Information (PII) with the string "REDACTED". The Advanced Auto-redact is an important feature for when you don't want full logs of sensitive customer information in their messages.

Author: It has been reworded since to what I've quoted below. What do you think @marklewin ?

 ```Auto-redact is automatic and is suitable for cases when you don't want sensitive customer information, like SMS message content, to end up in Vonage logs.```

> Vonage's Auto-redact feature enables customers to protect their SMS privacy. Auto-redact automatically replaces all Personal Identifiable Information (PII) with the string "REDACTED". The Advanced Auto-redact is an important feature for when you don't want full logs of sensitive customer information in their messages.

Reviewer: sounds good to me!

Businesses in industries that are highly regulated, such as financial services, education, and the police require redacted content to meet compliance and regulation rules. 

## How does Auto-redact work?

Redaction is the editing of a document or a record to delete or mask Personal Identifiable Information (PII) or confidential information. For example, your telephone number, bank details, or your address.

When you send SMS messages using the Vonage SMS API, two pieces of information containing PII get recorded: sender or receiver's phone number and the message body content.

> When a customer sends an SMS through the SMS API, logs get stored in two locations. Server logs get created as are transactional records of the activity known as a Call Detail Record (CDR). Server logs are retained for around 15 days, no more than one month. CDRs, however, are stored for 13 months. Both server logs and CDRs can be viewed by Vonage support staff, for testing, debugging, diagnosing user issues and reconciling CDRs against customer's transaction records.

Reviewer: According to this, logs are created in two locations, but we don't talk about _where_ they are stored ... aren't they really just two different types of log? And if, so, they both contain CDRs, so what's the difference? Are the server logs just a compilation of the individual CDRs?

Author: I've moved the diagram displaying the flow of the logs and CDRs to below this paragraph, which paints a better picture than a written description.

> When a customer sends an SMS through the SMS API, logs get stored in two locations. Server logs get created as are transactional records of the activity known as a Call Detail Record (CDR). Server logs are retained for around 15 days, no more than one month. CDRs, however, are stored for 13 months. Both server logs and CDRs can be viewed by Vonage support staff, for testing, debugging, diagnosing user issues and reconciling CDRs against customer's transaction records.

Author: Ah, this was rewritten.

> With Vonage, inbound messages can be auto-redacted with the Standard Auto-redact feature. For more information on the Standard Auto-redact feature, please refer to the [Redaction page](https://developer.nexmo.com/redact/overview).

Reviewer: ```suggestion
With Vonage, inbound messages can be auto-redacted with the Standard Auto-redact feature. For more information on the Standard Auto-redact feature, please refer to the [Redact API documentation](https://developer.nexmo.com/redact/overview).
```

> Advanced Auto-redact feature ensures that the message content is encrypted or redacted or both before the server logs and CDRs are stored. The phone number cannot be redacted in the server logs in order to meet data retention requirements. The configurations available are listed below:

Reviewer: ```suggestion
The Advanced Auto-redact feature ensures that the message content is encrypted or redacted or both before the server logs and CDRs are stored. The phone number cannot be redacted in the server logs in order to meet data retention requirements. The configurations available are listed below:
```

Reviewer: "The phone number cannot be redacted in the server logs in order to meet data retention requirements." This seems like an important point that is buried in the middle of this para ... can we call it out in the list of configurations?

Author: This has been improved on a fair bit making the points clearer.

> 

Reviewer: ```suggestion
The redaction options available are:

1. Message content redaction only.
2. Phone number redaction only (_but we've just said above this is not available, so perhaps we can clarify here?_)
3. Phone number encryption only.
4. Message content redaction combined with either redaction or encryption of the phone number.
```

Reviewer 2: Good question. Indeed, we have not described this in detail here. It is only described properly here: https://developer.nexmo.com/redact/overview#sms-api

When you see number redaction, it means several things:
1. If your redaction latency > 15 days then server logs have already expired thus any redaction you do to CDRs takes immediate effect. Of course, this redaction only touches the long-term storage, it does not touch meta information in the Legal Sequester (LS), but not all SMS CDRs end up in the Legal sequester. It has per-country filters. US SMS traffic is not captured in LS, for example. So, when redaction happens for US traffic, it is absolut. We do not publish the list of countries and their retention periods. We can give this information upon request. We are not supposed to advertise it so much.

2. redaction latency < 15 days, then what happens is that we encrypt the numbers first, so they get stored encrypted in server logs and CDRs. If the customer asked only for encryption (like Microsoft), we do not redact them further. However, if the customer asked, we do redact them further, but they are subject to data retention. So, we redact those encrypted numbers in the long-term storage. LS checks which of the encrypted numbers it should store, and stores them (it decrypts them and stores them). So, Redaction is possible because if you chose "encryption" the number will stay in the long-term storage in encrypted form for 13 months,  but with redaction it will not. So, immediate redaction always includes encryption first. Whether it stays in LS or not depends on the country.
The best tech description is here:
https://nexmoinc.atlassian.net/wiki/spaces/PSD/pages/1370423621/Automatic+Data+Redaction#AutomaticDataRedaction-SMS&NIDataflowandauto-redaction

1. Message content redaction only.
2. Phone number redaction only.
3. Phone number encryption only.
4. Message content redaction together with redaction or encryption of the phone number.

A key point of Vonage's immediate Advanced Auto-redact is that the message body is not stored in any of the logs or the CDRs. Phone numbers are also immediately encrypted, with an encryption key that Vonage staff do not have access to. Later those encrypted numbers get redacted as well. Please refer to the (Redaction page)[https://developer.nexmo.com/redact/overview] for more information.

The image below shows an example of the flow of Advanced SMS Auto-redact with Vonage.

![Advanced SMS Redaction](/assets/images/messaging/sms/advanced_sms_redaction.png)

## How to set up Auto-redact for your Vonage account

> You can find pricing relevant to the Advanced Auto-redact service on the [Vonage pricing](https://www.vonage.com/communications-apis/pricing/) page.

Reviewer: Perhaps we can swap these around so that we're showing how to set up first (as per the heading) and then linking to price data?
